### ISSUE TITLE
Tenant exists errors on migrate task on 2.8.0

### ISSUE THREAD

**Issue by jonian** [rails-on-services/apartment#136]
After updating to `2.8.0` I'm getting tenant exists errors when running `rails db:mirate`. I can see [here](https://github.com/rails-on-services/apartment/blob/89644421ef67e0005015c0a781e957050c263bb5/lib/tasks/apartment.rake#L34) that this behavior was added in the last version. Is this desired and we can ignore it?

If this is a desired behavior, can we create the schema if it does not exist and avoid throwing errors and executing `ROLLBACK` [here](https://github.com/rails-on-services/apartment/blob/276e5da69af0393d96e431a9f5d9e31896134ff9/lib/apartment/adapters/postgresql_adapter.rb#L99) by using:

```sql
CREATE SCHEMA IF NOT EXISTS "foo"

BEGIN;
CREATE SCHEMA IF NOT EXISTS "foo";
COMMIT;
```

PostgreSQL [docs](https://www.postgresql.org/docs/current/sql-createschema.html).

## Steps to reproduce
Update to version `2.8.0` and run the migration task:

```bash
rails db:migrate
```

## Expected behavior
Run migrations for the existing tenants.

## Actual behavior
Apartment tries to create the tenant before running the migrations and shows the output below:

```sql
Creating foo tenant
D, [2020-12-17T00:29:05.780156 #9] DEBUG -- :    (4.2ms)  BEGIN;
          CREATE SCHEMA "foo";
          COMMIT;
D, [2020-12-17T00:29:05.781050 #9] DEBUG -- :    (0.5ms)  ROLLBACK;
Tried to create already existing tenant: Error while creating tenant foo: PG::DuplicateSchema: ERROR:  schema "foo" already exists
```

## System configuration

* Database: PostgeSQL 11.6
* Apartment version: 2.8.0
* Apartment config (in `config/initializers/apartment.rb` or so):
  * `use_schemas`: (`true`)
* Rails (or ActiveRecord) version: 6.0.3.4
* Ruby version: 2.7.2

**Comment by rpbaltazar** [comment]
Thanks for reporting the issue. I think it makes sense what you're suggesting.
This all started with https://github.com/rails-on-services/apartment/issues/82 and has been cascading issues. I think i'll try to have your suggestion implemented and enhance the migrator with the ability to skip the creation if it already exists.

**Comment by jonian** [comment]
Thanks for the quick response @rpbaltazar! After thinking about this a bit more, maybe a config option, like the one below, is better than my previous suggestion.

```ruby
# The config key name and values can be something better named, just an example.

config.missing_tenant_on_migration = :raise_exception # default
config.missing_tenant_on_migration = :skip_migrations
config.missing_tenant_on_migration = :create_tenant
```

I think that apartment should not create tenants by default. It should raise an exception and leave it up to the application to handle the missing tenants. Maybe some applications already do that and this new behavior is a breaking change for them.

If you want, I can make a PR when you decide how you want this handled/implemented.

**Comment by rpbaltazar** [comment]
The configurable suggestion makes sense. I think that's what should have been done from the beginning. If you feel like digging in the gem source, I'm happy to take a look at the PR 👍

**Comment by rpbaltazar** [comment]
@jonian Just to be clear, will you be working on a PR for this, using a configurable behavior or shall i pick it up tomorrow?

**Comment by jonian** [comment]
@rpbaltazar I don't know if I will be able to complete this as fast as you, so if you have time for it then go ahead. If not, I can start working on it on Sunday. It's up to you.

**Comment by jonian** [comment]
@rpbaltazar I started working on this today.

**Comment by lizdeika** [comment]
There is a problem with `mysql` too. `ros-apartment` 2.8.1
Running `db:migrate` yields to:
```
Creating projectx_development tenant
rails aborted!
Configure a supported :charset and ensure innodb_large_prefix is enabled to support indexes on varchar(255) string columns.
bin/rails:4:in `require'
bin/rails:4:in `<main>'
Tasks: TOP => apartment:migrate
(See full trace by running task with --trace)
```

The problem was not present in 2.7.2
Commenting out `Apartment::TaskHelper.create_tenant(tenant)` helps too, so I figure tenant creation is the problem.
I guess it will be ok with PR#140 and `db_migrate_tenant_missing_strategy` set to `raise_exception` or `rescue_exception`

**Comment by rpbaltazar** [comment]
Resolved with #140

**Comment by paulomcnally** [comment]
Solved on 2.9.0. Thanks guys
